### ISSUE TITLE
Skip prereleases from "Releases" in PR descriptions

### ISSUE THREAD

**Issue by maciejmakowski** [dependabot/dependabot-core#4131]
While having the releases listed in Dependabot PR descriptions is great, prereleases do "pollute" that section with information that is not helpful.

Hence, it would be great if they could be filtered out.

Here's an example of a Dependabot PR with lots of prereleases (that we create for testing off feature branches):

![image](https://user-images.githubusercontent.com/1946435/129119105-c72ac25a-4efa-4e59-9e13-425d7dd114fb.png)

And for reference, one of those prereleases in the library "Releases" section:

![image](https://user-images.githubusercontent.com/1946435/129119263-4984d3b8-da64-4d51-826d-bf332342f914.png)

**Comment by asciimike** [comment]
Thanks for the feature request! Do you have a suggestion on how we could implement this in a sufficiently generic way (e.g. allow arbitrary filtering of release notes based on a regex) so that this isn't a one-off for you but a benefit for the entire user community?

**Comment by jojomatik** [comment]
Hi there, I'd also like to see this feature.

My suggestion would be to only show releases that are not tagged as "Pre-release"s. Usually the changelog for the next release contains the combined entries of all previous pre-releases. 

One could also omit all lines from pre-releases that are already mentioned in the following real release but that seems wrong to me.

Alternatively it would be possible to check if the releases follow a semver format and show only those that have no pre-release suffix (e.g. `-alpha` etc.) but only the three numbers `x.y.z` prefixed by an optional `v`. I believe dependabot is already using the semver spec to only return releases between the previous and new versions and no releases from another major version. The same spec could be used to determine pre-releases.

**Comment by jeffwidman** [comment]
👋 Sorry for the slow response.

The problem with this is that many open source projects will include their breaking changes in the notes for the alpha release of the breaking change, and then not mention them in the stable release, because they assume everyone can read the pre-release notes as the next bullet point in the changelog. TBH, I've done this myself on open source projects that were less popular when I've been short on time but want to keep the project moving forward.

This happens frequently enough across various popular open source projects that I just don't see us changing the Dependabot behavior.

It sounds like the problem is the Dependabot PR includes multiple of these pre-releases and is full of noise, so one possible workaround if these are just test builds is to add a CI action that deletes the previous release when it creates a new release. That way there's only a handful of pre-releases in the history between stable releases at any given time. So when Dependabot goes to create the new PR, it should be significantly less noisy with only 1-2 build-releases and the rest stable. May not work for your situation, just an idea.

**Comment by jeffwidman** [comment]
Another workaround is if you have two repos, the normal one for dev, and then a placeholder one that you use for stable releases and just pull the latest updates from dev and tag a stable release when you need. Clunky, but it'd probably work fine.

**Comment by jojomatik** [comment]
Thank you for the clarification. Although I'm not really happy with your solutions, I do understand why you can't change the behaviour globally.

Would it be possible to make that configurable though in `dependabot.yml` though @jeffwidman?

> Another workaround is if you have two repos, the normal one for dev, and then a placeholder one that you use for stable releases and just pull the latest updates from dev and tag a stable release when you need. Clunky, but it'd probably work fine.

This would indeed work, but as you said it's kind of clunky. Splitting (beta-)releases from the codebase does not seem right to me. Also the tooling I use to create releases (semantic-release) does not support this and I don't see a good way to implement it there. I'd still need the "version bump commits" in the main repository or they'd get out of sync really fast.

> It sounds like the problem is the Dependabot PR includes multiple of these pre-releases and is full of noise, so one possible workaround if these are just test builds is to add a CI action that deletes the previous release when it creates a new release. 

Similar situation here, this is not supported by the "standard" tools I use. Also I believe that changelogs should not be altered or deleted afterwards (except for typos etc.) but that might be my personal opinion only.

The "solution" for me will probably be to release less beta-versions. Currrently I release a beta-version for every change that benefit the end user (fixes, features, upstream dependency updates). Releasing beta-versions only once in a while should at least reduce the amount of headings. On the downside that means that beta users would receive less frequent updates and might need to build from source if they need a fix ASAP.

**Comment by jeffwidman** [comment]
We aren't going to add it as a config option in Dependabot run natively on GitHub.

You can also run `dependabot-core` yourself using an entry point like `dependabot-script`... Easy to wire up to run in the context of a GitHub Action. Since they're both open source you can fork and hack the code to meet your particular use case... That should also be a relatively straightforward solution.
